### PR TITLE
CAD-408: test buildability of `cabal new-build all`

### PR DESCRIPTION
**Pull request by deepfire**
1. Add a test that essentially builds everything `cardano-repo-tool` knows about at their current `master` commits:
```
      cardano-repo-tool clone-repos
      cabal new-configure --with-compiler ghc-8.6.5 -O0
      cabal new-build all
```
2. Add a `nix-shell` with `cabal-install` and `cardano-repo-tool` (aliased to `repo`).
3. This is wired at https://buildkite.com/input-output-hk/cardano-haskell/builds
4. Depends on https://github.com/input-output-hk/cardano-repo-tool/pull/6

***NOTE:*** The test currently fails, because of what appears to be an old Cabal version (2.4).  Cabal bump to V3 is tracked in CAD-414.

Work ongoing supplying Cabal v3 via `cardano-repo-tool` in https://github.com/input-output-hk/cardano-repo-tool/pull/6

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "cardano-repo-tool"]
 	path = cardano-repo-tool
-	url = https://github.com/input-output-hk/cardano-repo-tool.git
+	url = git@github.com:input-output-hk/cardano-repo-tool.git
```
Comment (CodiePP) [inline .gitmodules]: I think they want to keep the 'https://' method so anybody can download and execute it.

Comment (deepfire) [reply .gitmodules]: Fixed.

**Comment by deepfire** [comment]
> requires cabal3?

That is actually a fix -- Hamish did this to repair the problem with obtaining cabal-install.

**Comment by rvl** [comment]
@deepfire Check out #32 - it's based on the latest master branch.

**Comment by coot** [comment]
This seems to be a stale PR, so I am closing it; If you disagree please update the branch and re-open it.

**Pull request closed without merging**